packages/lexicon-resolver/src/record.ts

```diff
@@ -0,0 +1,125 @@
+import { CID } from 'multiformats/cid'
+import { IdResolver } from '@atproto/identity'
+import { RepoRecord } from '@atproto/lexicon'
+import {
+  Commit,
+  MST,
+  MemoryBlockstore,
+  def as repoDef,
+  readCarWithRoot,
+  verifyCommitSig,
+} from '@atproto/repo'
+import { AtUri, ensureValidDid } from '@atproto/syntax'
+import { BuildFetchHandlerOptions, FetchHandler } from '@atproto/xrpc'
+import { safeFetchWrap } from '@atproto-labs/fetch-node'
+import { AtpBaseClient as Client } from './client/index.js'
+import { isValidDid } from './util.js'
+
+export type ResolveRecordOptions = {
+  idResolver?: IdResolver
+  forceRefresh?: boolean
+  rpc?: Partial<BuildFetchHandlerOptions> | FetchHandler
+}
+
+export type RecordResolution = {
+  commit: Commit
+  uri: AtUri
+  cid: CID
+  record: RepoRecord
+}
+
+/**
+ * Resolve a record from the network, verifying its authenticity.
+ * @param uriStr AtUri or string representing one for the record that will be resolved.
+ * @param options
+ * @returns
+ */
+export async function resolveRecord(
+  uriStr: AtUri | string,
+  options: ResolveRecordOptions = {},
+): Promise<RecordResolution> {
+  const { idResolver = new IdResolver(), forceRefresh, rpc } = options
+  const uri = typeof uriStr === 'string' ? new AtUri(uriStr) : uriStr
+  const did = await getDidFromUri(uri, { idResolver })
+  const identity = await idResolver.did
+    .resolveAtprotoData(did, forceRefresh)
+    .catch((err) => {
+      throw new RecordResolutionError('Could not resolve DID identity data', {
+        cause: err,
+      })
+    })
+  const client = new Client(
+    typeof rpc === 'function'
+      ? rpc
+      : { service: identity.pds, fetch: safeFetch, ...rpc },
```

Comment:
I don't really like the spreading pattern when we want to apply defaults. It can be error prone:

If `rpc = { fetch: undefined }`, we won't get any typing error (because `fetch` is optional in `Partial<BuildFetchHandlerOptions> `) but we will end up *not* using safe fetch, as the `fetch: undefined` will end up overriding the `fetch: safeFetch` default.

I usually prefer enumerating properties (as this will allow better v8 optimizations), or, if we really want to spread:

```ts
{
  ...rpc,
  fetch: rpc.fetch ?? safeFetch,
  service: rpc.service ?? identity.pds,
}
```